### PR TITLE
Support Pricing Tier

### PR DESCRIPTION
**Pull request by 1openwindow**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

File: src/spring-cloud/azext_spring_cloud/_help.py [hunk 1]
```diff
@@ -316,3 +324,48 @@
     type: command
     short-summary: Update an Azure Cache for Redis service binding of the app.
 """
+
+helps['spring-cloud certificate add'] = """
```
Comment (yungezz) [inline src/spring-cloud/azext_spring_cloud/_help.py:328]: pls add examples for new commands.

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 2]
```diff
@@ -31,6 +31,13 @@ def validate_location(namespace):
                                       for piece in location_slice])
 
 
+
+def validate_sku(namespace):
+    namespace.sku = namespace.sku.upper()
+    if namespace.sku not in ['B0', 'S0']:
```
Comment (yungezz) [inline src/spring-cloud/azext_spring_cloud/_validators.py:37]: pls add supported value in help message

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 3]
```diff
@@ -31,8 +31,11 @@ def load_arguments(self, _):
             '--name', '-n'], help='Name of Azure Spring Cloud.')
 
     with self.argument_context('spring-cloud create') as c:
-        c.argument('location', arg_type=get_location_type(
-            self.cli_ctx), validator=validate_location)
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), validator=validate_location)
+        c.argument('sku', type=str, validator=validate_sku, help='Name of SKU, B0 for Basic, S0 for Standard')
```
Comment (qwordy) [inline src/spring-cloud/azext_spring_cloud/_params.py:35]: Use get_enum_type?

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 4]
```diff
@@ -190,3 +193,22 @@ def load_arguments(self, _):
                    validator=validate_app_name)
         c.argument('deployment', options_list=[
             '--deployment', '-d'], help='Name of an existing deployment of the app. Default to the production deployment if not specified.', validator=validate_deployment_name)
+
+    with self.argument_context('spring-cloud certificate') as c:
+        c.argument('service', service_name_type)
+        c.argument('name', name_type, help='Name of certificate.')
+
+    with self.argument_context('spring-cloud certificate add') as c:
+        c.argument('vault_uri', help='The key vault uri where store the certificate')
```
Comment (qwordy) [inline src/spring-cloud/azext_spring_cloud/_params.py:202]: Does it support vault name?

File: src/spring-cloud/azext_spring_cloud/_params.py [hunk 5]
```diff
@@ -190,3 +193,22 @@ def load_arguments(self, _):
...
+    with self.argument_context('spring-cloud app custom-domain') as c:
+        c.argument('service', service_name_type)
+        c.argument('app', app_name_type, help='Name of app.', validator=validate_app_name)
+        c.argument('domain_name', help='Name of custom domain.')
+
+    with self.argument_context('spring-cloud app custom-domain bind') as c:
+        c.argument('certificate', type=str, help='Certificate name in Azure Spring Cloud.')
+
+    with self.argument_context('spring-cloud app custom-domain update') as c:
+        c.argument('certificate', help='Certificate name in Azure Spring Cloud.')
```
Comment (qwordy) [inline src/spring-cloud/azext_spring_cloud/_params.py:214]: One has `type=str`, while the other does not.

File: src/spring-cloud/azext_spring_cloud/_utils.py [hunk 6]
```diff
@@ -193,3 +193,11 @@ def _get_rg_location(ctx, resource_group_name, subscription_id=None):
     # Just do the get, we don't need the result, it will error out if the group doesn't exist.
     rg = groups.get(resource_group_name)
     return rg.location
+
+def _get_sku_tier(name):  # pylint: disable=too-many-return-statements
+    name = name.upper()
+    if name in ['B0']:
```
Comment (qwordy) [inline src/spring-cloud/azext_spring_cloud/_utils.py:199]: What's the benefit of `name in ['B0']` compared to `name == 'B0'`?

File: src/spring-cloud/azext_spring_cloud/_validators.py [hunk 7]
```diff
@@ -31,6 +31,13 @@ def validate_location(namespace):
                                       for piece in location_slice])
 
 
+
+def validate_sku(namespace):
+    namespace.sku = namespace.sku.upper()
+    if namespace.sku not in ['B0', 'S0']:
+        raise CLIError("The pricing tier include B0(Basic), S0(Standard), only accept value [B0, S0]")
```
Comment (qwordy) [inline src/spring-cloud/azext_spring_cloud/_validators.py:38]: Grammar error.

File: src/spring-cloud/azext_spring_cloud/commands.py [hunk 8]
```diff
@@ -81,5 +86,18 @@ def load_command_table(self, _):
         g.custom_command('redis update', 'binding_redis_update')
         g.custom_show_command('remove', 'binding_remove')
 
+    with self.command_group('spring-cloud certificate', client_factory=cf_certificates) as g:
+        g.custom_command('add', 'certificate_add')
+        g.custom_command('show', 'certificate_show', table_transformer=transform_spring_cloud_certificate_output)
```
Comment (qwordy) [inline src/spring-cloud/azext_spring_cloud/commands.py:91]: custom_show_command

File: src/spring-cloud/azext_spring_cloud/commands.py [hunk 9]
```diff
@@ -81,5 +86,18 @@ def load_command_table(self, _):
...
 
+    with self.command_group('spring-cloud certificate', client_factory=cf_certificates) as g:
+        g.custom_command('add', 'certificate_add')
+        g.custom_command('show', 'certificate_show', table_transformer=transform_spring_cloud_certificate_output)
+        g.custom_command('list', 'certificate_list', table_transformer=transform_spring_cloud_certificate_output)
+        g.custom_command('remove', 'certificate_remove')
+
+    with self.command_group('spring-cloud app custom-domain', client_factory=cf_custom_domains) as g:
+        g.custom_command('bind', 'domain_bind')
+        g.custom_command('show', 'domain_show', table_transformer=transform_spring_cloud_custom_domain_output)
```
Comment (qwordy) [inline src/spring-cloud/azext_spring_cloud/commands.py:97]: custom_show_command

File: src/spring-cloud/azext_spring_cloud/commands.py [hunk 10]
```diff
@@ -4,15 +4,20 @@
...
 
 
 # pylint: disable=too-many-statements
 def load_command_table(self, _):
     with self.command_group('spring-cloud', client_factory=cf_app_services) as g:
         g.custom_command('create', 'spring_cloud_create',
+                         supports_no_wait=True),
+        g.custom_command('update', 'spring_cloud_update',
                          supports_no_wait=True)
         g.custom_command('delete', 'spring_cloud_delete',
```
Comment (qwordy) [inline src/spring-cloud/azext_spring_cloud/commands.py:22]: Maybe you can add a `confirmation=True`

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 11]
```diff
@@ -1,1079 +1,1159 @@
...
+LOG_RUNNING_PROMPT = "This command usually takes minutes to run. Add '--verbose' parameter if needed."
+
+
+def spring_cloud_create(cmd, client, resource_group, name, location=None, sku=None, no_wait=False):
+    rg_location = _get_rg_location(cmd.cli_ctx, resource_group)
+    if location is None:
+        location = rg_location
+
+    if sku is None:
+        sku = "S0"
```
Comment (fengzhou-msft) [inline src/spring-cloud/azext_spring_cloud/custom.py:47]: You can set `default='S0'` in `_params.py` when defining `sku` in `spring-cloud create`.

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 12]
```diff
@@ -1,1079 +1,1159 @@
...
+
+    full_sku = models.Sku(
+        name=sku,
+        tier=_get_sku_tier(sku)
+    )
```
Comment (fengzhou-msft) [inline src/spring-cloud/azext_spring_cloud/custom.py:52]: This model construction and the `_get_sku_tier` logic could both be moved to the `validate_sku` validator if you remove `type=str` in `_params.py`. Then here you would get the `sku` parameter directly as a `models.Sku`. It's fine, just providing another common design choice in CLI.

File: src/spring-cloud/azext_spring_cloud/custom.py [hunk 13]
```diff
@@ -1,1079 +1,1159 @@
...
+    return sdk_no_wait(no_wait, client.update,
+                       resource_group_name=resource_group, service_name=name, resource=resource)
+
+def spring_cloud_delete(cmd, client, resource_group, name, no_wait=False):
+    return sdk_no_wait(no_wait, client.delete,
+                       resource_group_name=resource_group, service_name=name)
+
+
+def spring_cloud_list(cmd, client, resource_group=None):
+    if resource_group is None:
```
Comment (fengzhou-msft) [inline src/spring-cloud/azext_spring_cloud/custom.py:75]: We allow passing an empty group to bypass the default group value set with `az configure -d group=Value`.
```suggestion
    if not resource_group:
```

File: src/spring-cloud/azext_spring_cloud/commands.py [hunk 14]
```diff
@@ -81,5 +86,18 @@ def load_command_table(self, _):
...
+        g.custom_command('show', 'certificate_show', table_transformer=transform_spring_cloud_certificate_output)
+        g.custom_command('list', 'certificate_list', table_transformer=transform_spring_cloud_certificate_output)
+        g.custom_command('remove', 'certificate_remove')
+
+    with self.command_group('spring-cloud app custom-domain', client_factory=cf_custom_domains) as g:
+        g.custom_command('bind', 'domain_bind')
+        g.custom_command('show', 'domain_show', table_transformer=transform_spring_cloud_custom_domain_output)
+        g.custom_command('list', 'domain_list', table_transformer=transform_spring_cloud_custom_domain_output)
+        g.custom_command('update', 'domain_update')
+        g.custom_command('unbind', 'domain_unbind')
```
Comment (fengzhou-msft) [inline src/spring-cloud/azext_spring_cloud/commands.py:100]: In `_help.py`, it's called `delete`. Please make it consistent and run `azdev linter spring-cloud` to validate.

**Pull request closed without merging**